extends Solid


func _ready():
	object_setup()


func _physics_process(_delta):
	object_tick()


func _process(delta):
	object_process(delta)


#--- Object functions


func destroy():
	# action_inherited
	super()
	
	# main_code
	if (not clean_death and not global.clean_solids):

		var rubble = gml.instance_create(position.x+8+randi_range(0,8)-randi_range(0,8), position.y+8+randi_range(0,8)-randi_range(0,8), "rubble")
		rubble.sprite_index = "rubble_tan"
		rubble = gml.instance_create(position.x+8+randi_range(0,8)-randi_range(0,8), position.y+8+randi_range(0,8)-randi_range(0,8), "rubble_small")
		rubble.sprite_index = "rubble_tan_small"
		rubble = gml.instance_create(position.x+8+randi_range(0,8)-randi_range(0,8), position.y+8+randi_range(0,8)-randi_range(0,8), "rubble_small")
		rubble.sprite_index = "rubble_tan_small"
